client/components/domains/use-my-domain/index.jsx

```diff
@@ -80,13 +78,13 @@ function UseMyDomain( { goBack, initialQuery, selectedSite } ) {
 
 				if ( availabilityErrorMessage ) {
 					setDomainNameValidationError( availabilityErrorMessage );
-					return;
+				} else {
+					setMode( inputMode.transferOrConnect );
+					setDomainAvailabilityData( availabilityData );
 				}
-
-				setMode( inputMode.transferOrConnect );
-				setDomainAvailabilityData( availabilityData );
-			}
-		);
+			} )
+			.catch( ( error ) => setDomainNameValidationError( error ) )
```

Comment:
Very nitpicky, but if only `error` is being used, you can simplify it:

```suggestion
			.catch( setDomainNameValidationError )
```